ADBench/run-all.ps1

```diff
@@ -53,23 +53,51 @@ param(# Which build to test.
       [double]$timeout=300,    
       
       # Where to store the ouput, defaults to tmp/ in the project root 
-      [string]$tmpdir="",      
+      [string]$tmpdir="",
       
       # Repeat tests, even if output file exists
-      [switch]$repeat,         
+      [switch]$repeat,
       
       # Repeat only failed tests
       [switch]$repeat_failures,
       
       # List of tools to run
-      [string[]]$tools=@(),    
+      [string[]]$tools=@(),
+      
+      # Don't delete produced jacobians even if they're accurate
+      [switch]$keep_correct_jacobians, 
       
       # GMM D values to try.  Must be a subset of the list of
       # compiled values in ADBench/cmake-vars-$buildtype.ps1                         
       [int[]]$gmm_d_vals_param,
       
       # GMM K values to run.  As above.
-      [int[]]$gmm_k_vals_param 
+      [int[]]$gmm_k_vals_param,
+
+      # GMM sizes to try. Must be a subset of @("1k", "10k", "2.5M")
+      # 2.5M currently is not supported
+      [string[]]$gmm_sizes = @("1k", "10k"),
+
+      # Hand problem sizes to try. Must be a subset of @("small", "big")
+      [string[]]$hand_sizes = @("small", "big"),
+
+      # Number of the first BA problem to try. Must be between 1 and ba_max_n
+      [int]$ba_min_n = 1,
+
+      # Number of the last BA problem to try. Must be between ba_min_n and 20
+      [int]$ba_max_n = 5,
```

Comment:
It would seem more natural to me if this was the largest possible (which seems to be 20).  Is there a good reason the default wouldn't be that?

Reply:
(I see that the original default was 5.  If the max possible is 20 wouldn't that be a better default?  Maybe we should leave that for another PR though.)

Reply:
(I never knew that BA and HAND could be bigger and I'm quite surprised to discover it!)

Reply:
While the best person to explain the justification behind the current defaults is, probably, @awf , here're some of my thoughts on the matter:

20 would be a bad default for BA, because that's the one where just the size of the output Jacobian is around 20 GB. That would simply crash most tools on most machines.
Even before that the sizes of the result quickly become very big to the point, where just the time to save them could cause the timeout. So, it's important to balance this default and the default for timeout.

We also need to balance these defaults against the time needed to complete the default set of benchmarks. It's two days as it is.

I think, it would be best to raise an issue about the defaults, and change them in a separately, if we decide to do so.

Anyway, the defaults are not the topic of this PR.

Reply:
> Anyway, the defaults are not the topic of this PR.

Agreed